### PR TITLE
clc patch

### PR DESCRIPTION
**Pull request by aspannaus**
Update clc classes, methods, and such from enclave dev version.

### CODE REVIEW

File: fresco/abstention/abstention.py
Comment (aspannaus) [inline fresco/abstention/abstention.py:1]: - correct scaling for min/max alphas
- adjust modify_alpha methods for correct behavior when micro == 0
-

File: fresco/data_loaders/data_utils.py
Comment (aspannaus) [inline fresco/data_loaders/data_utils.py:1]: - edit dataloaders for intended clc doc embeds creation
- add GroupedCases class from enclave
-

File: fresco/models/mthisan.py
Comment (aspannaus) [inline fresco/models/mthisan.py:1]: add padding_idx on hisan definition for consistency with cnn

File: fresco/predict/predictions.py
Comment (aspannaus) [inline fresco/predict/predictions.py:1]: correct device, gpu vs cpu for getting ys with clc

File: fresco/run_clc.py
Comment (aspannaus) [inline fresco/run_clc.py:1]: updated from enclave version

- updated base model loading from improved checkpointing
- improved creating doc_embeds
- allow for base cnn model in clc
-

File: fresco/validate/validate_params.py
Comment (aspannaus) [inline fresco/validate/validate_params.py:1]: check for ntask and dac earlier to check for ntask == True but dac == False

File: setup.py
Comment (aspannaus) [inline setup.py:1]: bump version number
add polars to deps

**Review by aspannaus — COMMENTED** [review]
changes from enclave clc version

File: fresco/data_loaders/data_utils.py [hunk 1]
```diff
@@ -338,16 +339,18 @@ def load_weights(self, fold):
...
         """
         Create torch DataLoader classes for training module.
 
         Returns a dictionary of PyTorch DataLoaders (train, val, test) for the training module.
 
         Args:
-            unk_tok (int): Token to convert unknown words to.
-            vocab_size (int): Number of words in the vocabulary.
             switch_rate (float): Proportion of words in each document to randomly flip.
+            reproducible (bool): Set all random number seeds.
```
Comment (zachfox) [inline fresco/data_loaders/data_utils.py:351]: does `reproducible = True` set `shuffle_data=False`? or is it that if `reproducible = True`, you always get the same shuffling of the data?

File: fresco/data_loaders/data_utils.py [hunk 2]
```diff
@@ -377,16 +389,11 @@ def make_torch_dataloaders(self, switch_rate: float,
                                  tasks=self.model_args['data_kwargs']['tasks'],
                                  label_encoders=self.dict_maps['id2label'],
                                  max_len=self.model_args['train_kwargs']['doc_max_len'],
-                                 transform=AddNoise(unk_tok,
-                                                    self.model_args['train_kwargs']['doc_max_len'],
-                                                    vocab_size,
-                                                    switch_rate,
-                                                    seed
-                                                    )
+                                 transform=_transform
```
Comment (zachfox) [inline fresco/data_loaders/data_utils.py:392]: very nice

File: fresco/data_loaders/data_utils.py (hunk 1 above)
Comment (aspannaus) [reply fresco/data_loaders/data_utils.py:351]: Setting `reproducible = True` gets you the same shuffling of the data. If we want to have the ability to turn off shuffling, I can add that as a keyword arg. Thoughts?

Comment (zachfox) [reply fresco/data_loaders/data_utils.py:351]: I think the current behavior is in line with what a user would expect. `reproducible=False` and `shuffle_data=True` would give shuffling, so I don't think we need an additional keyword?

Comment (aspannaus) [reply fresco/data_loaders/data_utils.py:351]: I was thinking of the case where we would want to turn off shuffling of the data in all instances. Adding an additional keyword arg would help with that case.

**Review by zachfox — APPROVED** [review]
Changes look good. I tested CLC and found a small bug which was addressed.

**Pull request merged**